controllers/machineset_controller.go

```diff
@@ -618,6 +618,7 @@ func (r *MachineSetReconciler) updateStatus(ctx context.Context, cluster *cluste
 	fullyLabeledReplicasCount := 0
 	readyReplicasCount := 0
 	availableReplicasCount := 0
+	deletingReplicasCount := 0
```

Comment:
While this is a fine change, I'd prefer to have a different PR for it, given that we don't necessarily need it to make the `OnDelete` strategy work, wdyt?

Reply:
@vincepri we do actually need it to work:

Basically that field is what the MachineDeployment Controller uses this field during reconciliation of `Old MachineSets` to determine what the proper MachineSet Replica count should be when a node is removed.

Reply:
We usually try to avoid using Status fields to inform reconciliation steps, especially if these change during reconciliations. Would you be open to have a small utility function to calculate this value by passing a MachineSet in it or something similar to that? 

Then we can use that value to inform status as well, but not the other way around.

Reply:
Yea! We can do anything..... My main concern was "avoiding" the MachineDeployment looking at anything further down from the MachineSet abstraction if that makes sense. Effectively to do what you want: Just during reconciliation of the MachineDeployment we would also need to fetch the machines associated with the old MachineSet and then just move the calculation there. 

Is the tradeoff better in your eyes that we now have the MachineDeployment controller also querying machines and no longer *just* looking at machinesets? If so we will go with that approach

Basically my goal was to keep the MachineSet controller the only controller looking at downstream machines  (besides the MachineController itself)
And then MachineDeployments only looking at MachineSet fields to have that clear laying.

And then feedback sort of goes up from there if that makes sense. But to be clear we could remove the field by effectively on every old machine set with replicas querying for deleting nodes and getting the values at that point in time.

I believe it will result in more network calls though if that matters to you